resources/test/fixtures/E741.py

```diff
@@ -0,0 +1,12 @@
+O = 0
+I = 0
+l = 0
+O: int = 0
+
+a, O = 0, 1
+[a, O] = 0, 1
+a, *O = 0, 1, 2
+a = O = 0
+
+o = 0
+i = 0
```

Comment:
Looks like the following cases are not covered:

```
for I in range(3):
    pass

with ctx() as I:
  pass

nonlocal I
global I
```

Reply:
https://github.com/PyCQA/pycodestyle/blob/10a4427c75740717b43448339fcf71f11bc33d1a/pycodestyle.py#L1482

```
@register_check
def ambiguous_identifier(logical_line, tokens):
    r"""Never use the characters 'l', 'O', or 'I' as variable names.

    In some fonts, these characters are indistinguishable from the
    numerals one and zero. When tempted to use 'l', use 'L' instead.

    Okay: L = 0
    Okay: o = 123
    Okay: i = 42
    E741: l = 0
    E741: O = 123
    E741: I = 42

    Variables can be bound in several other contexts, including class
    and function definitions, 'global' and 'nonlocal' statements,
    exception handlers, and 'with' and 'for' statements.
    In addition, we have a special handling for function parameters.

    Okay: except AttributeError as o:
    Okay: with lock as L:
    Okay: foo(l=12)
    Okay: for a in foo(l=12):
    E741: except AttributeError as O:
    E741: with lock as l:
    E741: global I
    E741: nonlocal l
    E741: def foo(l):
    E741: def foo(l=12):
    E741: l = foo(l=12)
    E741: for l in range(10):
    E742: class I(object):
    E743: def l(x):
    """
```